### PR TITLE
removin userid

### PR DESCRIPTION
**Pull request by tinakurian**
removing userid to make registration service build using latest toolchain-api

### CODE REVIEW

File: pkg/signup/signup_service.go [hunk 1]
```diff
@@ -97,7 +97,6 @@ func (s *ServiceImpl) CreateUserSignup(username, userID string) (*crtapi.UserSig
 			Namespace: s.Namespace,
 		},
 		Spec: crtapi.UserSignupSpec{
-			UserID:        userID,
```
Comment (alexeykazakov) [inline pkg/signup/signup_service.go]: We still need userID. But now it's used as UserSignup.matadata.name.

File: go.sum [hunk 2]
```diff
@@ -1,8 +1,21 @@
+bitbucket.org/ww/goautoneg v0.0.0-20120707110453-75cd24fc2f2c/go.mod h1:1vhO7Mn/FZMgOgDVGLy5X1mE6rq1HbkBdkF/yj8zkcg=
```
Comment (alexeykazakov) [inline go.sum]: Please do `go mod tidy`.

Comment (tinakurian) [reply go.sum]: fixed here: ebd346dbee0e7459a4ac41f1332b5766178867b5

File: pkg/signup/signup_service.go (hunk 1 above)
Comment (tinakurian) [reply pkg/signup/signup_service.go]: Hm.. is this part of the spec? i don't see it here: https://github.com/codeready-toolchain/api/blob/master/pkg/apis/toolchain/v1alpha1/usersignup_types.go#L19. 

Does this mean we will still be using it to create a usersignup?

Comment (alexeykazakov) [reply pkg/signup/signup_service.go]: It's not in Spec. It's resource name - https://kubernetes.io/docs/concepts/overview/working-with-objects/names/
It's in metadata - https://github.com/codeready-toolchain/api/blob/master/pkg/apis/toolchain/v1alpha1/usersignup_types.go#L74

**Pull request closed without merging**